From: Daniel Mierswa <impulze@impulze.org>
Date: Thu, 3 Feb 2011 03:36:13 +0000 (+0100)
Subject: backport multi tags patch
X-Git-Url: http://git.videolan.org/?p=vlc%2Fvlc-1.1.git;a=commitdiff_plain;h=8f350488599b292521f4d460a945f13712fa6c75

backport multi tags patch

based on vlc.git 7be936ca451fa0c052069895460dcc0e67f4dfa3

Signed-off-by: Jean-Baptiste Kempf <jb@videolan.org>
---

diff --git a/modules/demux/mkv/matroska_segment.cpp b/modules/demux/mkv/matroska_segment.cpp
index 117dc03..25c9971 100644
--- a/modules/demux/mkv/matroska_segment.cpp
+++ b/modules/demux/mkv/matroska_segment.cpp
@@ -220,6 +220,7 @@ void matroska_segment_c::LoadTags( KaxTags *tags )
                     }
                     ep->Up();
                 }
+#if LIBMATROSKA_VERSION < 0x010100
                 else if( MKV_IS_ID( el, KaxTagGeneral ) )
                 {
                     msg_Dbg( &sys.demuxer, "|   + General" );
@@ -288,6 +289,7 @@ void matroska_segment_c::LoadTags( KaxTags *tags )
                 {
                     msg_Dbg( &sys.demuxer, "|   + Multi Title" );
                 }
+#endif
                 else
                 {
                     msg_Dbg( &sys.demuxer, "|   + LoadTag Unknown (%s)", typeid( *el ).name() );
diff --git a/modules/demux/mkv/mkv.hpp b/modules/demux/mkv/mkv.hpp
index 8b89b78..99a407c 100644
--- a/modules/demux/mkv/mkv.hpp
+++ b/modules/demux/mkv/mkv.hpp
@@ -92,13 +92,15 @@
 #include "matroska/KaxSegment.h"
 #include "matroska/KaxTag.h"
 #include "matroska/KaxTags.h"
+#include "matroska/KaxVersion.h"
+#if LIBMATROSKA_VERSION < 0x010100
 #include "matroska/KaxTagMulti.h"
+#endif
 #include "matroska/KaxTracks.h"
 #include "matroska/KaxTrackAudio.h"
 #include "matroska/KaxTrackVideo.h"
 #include "matroska/KaxTrackEntryData.h"
 #include "matroska/KaxContentEncoding.h"
-#include "matroska/KaxVersion.h"
 
 #include "ebml/StdIOCallback.h"
 
